Products/CMFCore/subscribers.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+"""
+Subscriber Support
+------------------
+
+The package comes with support for queueing up and performing indexing
+operations via event subscribers.  The idea behind this is to not rely on
+explicit calls as defined in ``CMFCatalogAware`` alone, but instead make it
+possible to phase them out eventually. As the additional indexing operations
+added via the subscribers are optimized away anyway, this only adds very
+little processing overhead.
+
+However, even though ``IObjectModifiedEvent`` has support for partial
+reindexing by passing a list of descriptions/index names, this is currently
+not used anywhere in Plone. Unfortunately that means that partial reindex
+operations will be "upgraded" to full reindexes, e.g. for
+``IContainerModifiedEvent`` via the ``notifyContainerModified`` helper,
+which is one reason why subscriber support is not enabled by default for now.
+
+To activate please use::
+
+    [instance]
+    ...
+    zcml = Products.CMFCore:subscribers.zcml
+
+instead of just the package name itself, re-run buildout and restart your
+Plone instance.
+"""
+from Products.CMFCore.indexing import filterTemporaryItems
+from Products.CMFCore.indexing import getQueue
+from zope.container.contained import dispatchToSublocations
+from zope.event import notify
+from zope.lifecycleevent import ObjectModifiedEvent, Attributes
+
+
+def objectAdded(ev):
+    obj = filterTemporaryItems(ev.object)
+    indexer = getQueue()
+    if obj is not None and indexer is not None:
+        indexer.index(obj)
```

Comment:
`indexer` cannot be None, unless you adopt the suggestion above that `getQueue` check the environment variable.